src/analysis/ml.py

```diff
@@ -10,14 +9,15 @@
 from sklearn.cluster import AgglomerativeClustering
 from sklearn.decomposition import PCA
 from sklearn.preprocessing import StandardScaler
-
+from adjustText import adjust_text
 from src.util import make_required_dirs
 
 plt.switch_backend('Agg')
 
+linkage_methods = ["ward", "complete", "average", "single"]
+distance_metrics = ["euclidean", "l1", "l2", "manhattan", "cosine", "precomputed"]
```

Comment:
Probably need to remove `precomputed`. Making these sets is slightly more efficient if we are checking membership and not using the order.